db/schema.sql

```diff
@@ -0,0 +1,103 @@
+SELECT count(*) FROM information_schema.tables WHERE table_schema = CURRENT_SCHEMA() AND table_name = 'migrations' AND table_type = 'BASE TABLE';
+
+CREATE TABLE "migrations" (
+    "id" varchar(255),
+    PRIMARY KEY ("id")
+);
+
+CREATE TABLE "tenants" (
+    "created_at" timestamptz NOT NULL,
+    "updated_at" timestamptz,
+    "deleted_at" timestamptz,
+    "tenant_id" serial,
+    "tenant_storage_contract_cid" text NOT NULL,
+    "tenant_meta" jsonb NOT NULL DEFAULT '{}',
+    "tenant_settings" jsonb NOT NULL DEFAULT '{}',
+    PRIMARY KEY ("tenant_id")
+);
+
+CREATE INDEX IF NOT EXISTS "idx_tenants_deleted_at" ON "tenants" ("deleted_at");
+
+CREATE TABLE "tenants_sps" (
+    "tenant_id" integer,
+    "sp_id" integer,
+    "tenant_sp_state" tenant_sp_state NOT NULL DEFAULT 'eligible',
+    "tenant_sps_meta" jsonb NOT NULL DEFAULT '{}',
+    PRIMARY KEY ("tenant_id", "sp_id")
+);
+
+CREATE TABLE "addresses" (
```

Comment:
No unique constraint - desired?

Reply:
There is this unique constraint below:

`CREATE UNIQUE INDEX IF NOT EXISTS "idx_address_tenant_id" ON "addresses" ("tenant_id", "address");` 

so each pair of `tenant_id` and `address` must be unique, but multiple tenants could have the same address if they wanted.